controllers/cloud.redhat.com/namespacereservation_controller.go

```diff
@@ -67,7 +67,7 @@ func (r *NamespaceReservationReconciler) Reconcile(ctx context.Context, req ctrl
 			// Must have been deleted
 			return ctrl.Result{}, nil
 		}
-		r.log.Error(err, "Reservation Not Found")
+		r.log.Error(err, fmt.Sprintf("Reservation not found for namespace [%s]", req.NamespacedName.Namespace))
```

Comment:
OK, so we established that it's not a "NotFound" error, but then why are we emitting the log message that says it's not found, outside of the "NotFound" block. this log message will fire saying that the Reservation is not found, but in actuality the error is going to be anything but Not Found ;)